tests/alley/wp/alleyvate/features/test-disable-comments.php

```diff
@@ -0,0 +1,318 @@
+
+	/**
+	 * Feature instance.
+	 *
+	 * @var Feature
+	 */
+	private Feature $feature;
+
+	/**
+	 * Set up.
+	 */
+	protected function setUp(): void {
+		parent::setUp();
+
+		// Turn off comment flood checking in order to run these tests.
+		add_filter( 'wp_is_comment_flood', '__return_false', \PHP_INT_MAX );
+
+		$this->feature = new Disable_Comments();
+	}
+
+	/**
+	 * Tear down.
+	 */
+	protected function tearDown(): void {
+		parent::tearDown();
+
+		// Turn on comment flood checking again.
+		remove_filter( 'wp_is_comment_flood', '__return_false', \PHP_INT_MAX );
+	}
+
+	/**
+	 * Test that the feature prevents fetching a count of comments via the get_comments function.
+	 */
+	public function test_get_comments_count_returns_empty(): void {
+		$post_id = self::factory()->post->create();
+		wp_insert_comment(
+			[
+				'comment_post_ID' => $post_id,
+				'comment_content' => 'Lorem ipsum dolor sit amet.',
+			]
+		);
+		$this->assertSame(
+			1,
+			get_comments(
+				[
+					'post_id' => $post_id,
+					'count'   => true,
+				]
+			)
+		);
+
+		// Activate the disable comments feature.
+		$this->feature->boot();
+
+		$this->assertSame(
+			0,
+			get_comments(
+				[
+					'post_id' => $post_id,
+					'count'   => true,
+				]
+			)
+		);
+	}
+
+	/**
+	 * Test that the feature prevents posting new comments.
+	 */
+	public function test_prevent_comment_posting(): void {
+		$post_id = self::factory()->post->create();
+
+		// Post a comment on the post and ensure that it posts correctly.
+		$result_pre = wp_handle_comment_submission(
+			[
+				'author'          => 'Test Author',
+				'comment'         => 'Lorem ipsum dolor sit amet.',
+				'comment_parent'  => 0,
+				'comment_post_ID' => $post_id,
+				'email'           => 'user@example.org',
+				'url'             => 'https://example.org',
+			]
+		);
+		$this->assertNotWPError( $result_pre );
+
+		// Activate the disable comments feature.
+		$this->feature->boot();
+
+		// Try again, and this time it should fail to insert.
+		$result_post = wp_handle_comment_submission(
+			[
+				'author'          => 'Testy McTesterson',
+				'comment'         => 'A new comment on a post with comments closed.',
+				'comment_parent'  => 0,
+				'comment_post_ID' => $post_id,
+				'email'           => 'testy@example.org',
+				'url'             => 'https://example.org/testy',
+			]
+		);
+		$this->assertWPError( $result_post );
+	}
+
+	/**
+	 * Test that the Comments menu item is removed from the primary admin menu on the left and the admin bar.
+	 */
+	public function test_remove_comments_from_admin_menus(): void {
+		// Reset admin menus.
+		/* phpcs:disable WordPress.WP.GlobalVariablesOverride.Prohibited */
+		global $menu, $submenu;
+		$menu    = [];
+		$submenu = [];
+		/* phpcs:enable */
+
+		// Become admin and load the admin menu to build the $menu global.
+		$this->acting_as( 'administrator' );
+		require ABSPATH . 'wp-admin/includes/plugin.php';
+		require ABSPATH . 'wp-admin/menu.php';
```

Comment:
just in case mantle starts to load them as part of `Admin_Screen`
```suggestion
		require_once ABSPATH . 'wp-admin/includes/plugin.php';
		require_once ABSPATH . 'wp-admin/menu.php';
```